### PR TITLE
[Prow CI workflow Enablement] Use HyperShift cli create HostedCluster on BM + KubeVirt platform

### PR DESCRIPTION
**Pull request by LiangquanLi930**
This PR creates workflow: `hypershift-kubevirt-baremetalds-conformance`, including odf, metallb and kubevirt, etc

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ chain:
       oc wait --timeout=30m --for=condition=Available --namespace=clusters hostedcluster/${CLUSTER_NAME}
       echo "Cluster became available, creating kubeconfig"
       bin/hypershift create kubeconfig --name=${CLUSTER_NAME} >${SHARED_DIR}/nested_kubeconfig
+      
+      if [ "$(oc get infrastructure cluster -o=jsonpath='{.status.platformStatus.type}')" == "BareMetal" ]; then
+        exit 0
+      fi
```
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:37]: why not add the proxy to this script within this if/def?

I know it's more code, but in this case i think making this one script more complicated is actually simpler then the indirection caused by breaking out of this script earlier and executing multiple other steps for the proxy and health check.

One improvement now that this script is so large would be to break this script into a `.sh` file that's referenced by the yaml file.

File: ci-operator/step-registry/hypershift/kubevirt/baremetalds/proxy/hypershift-kubevirt-baremetalds-proxy-commands.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+sudo podman run -d --rm \
+     --net host \
+     --volume \$HOME/squid.conf:/etc/squid/squid.conf \
+     --name external-squid \
+     --dns 127.0.0.1 \
+     quay.io/openshifttest/squid-proxy:multiarch
+EOF
+
+echo "Adding proxy-url in kubeconfig"
+sed -i "/- cluster/ a\    proxy-url: http://$IP:8213/" "${SHARED_DIR}"/nested_kubeconfig
```
Comment (davidvossel) [inline ci-operator/step-registry/hypershift/kubevirt/baremetalds/proxy/hypershift-kubevirt-baremetalds-proxy-commands.sh:26]: can we just add this and the corresponding health check inline to the `hypershift kubevirt create chain`?

**Review by davidvossel — COMMENTED** [review]
in general, this makes sense to me. I left one minor comment about the guest cluster creation flow.

Other than that, it would be nice to see the rehersals pass before merging

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:37]: Yes, I would love to too, but AFAIK there is no ssh in this image.

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml (hunk 1 above)
Comment (davidvossel) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:37]: ah, ok that makes sense. i understand

File: ci-operator/step-registry/hypershift/kubevirt/baremetalds/proxy/hypershift-kubevirt-baremetalds-proxy-commands.sh (hunk 2 above)
Comment (davidvossel) [reply ci-operator/step-registry/hypershift/kubevirt/baremetalds/proxy/hypershift-kubevirt-baremetalds-proxy-commands.sh:26]: nevermind, that's impractical at the moment due to the image not having ssh

**Review by davidvossel — COMMENTED** [review]
/approve

lets see the periodics pass before merging

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**